src/ast/spans.rs

```diff
@@ -2280,6 +2277,12 @@ impl Spanned for TableObject {
     }
 }
 
+impl Spanned for BeginEndStatements {
+    fn span(&self) -> Span {
+        union_spans([self.begin_token.0.span, self.end_token.0.span].into_iter())
```

Comment:
We can do an exhausitve match here, so that the code is revisite if fields get added or changed
```rust
let BeginEndStatements {
    begin,
    statements,
    end
} = self;
union_spans(begin, statements, end)
```
we can probably include statements in the union spans call if only for completeness

Reply:
I updated this, although I'm unsure if my implementation was what you're suggesting